web/lumpTypeView.ts

```diff
@@ -478,97 +484,238 @@ export function LumpTypeViewPlaypal(scaleX:number = 4, scaleY:number = 4) {
+                    drawPalette(currentPaletteIndex);
                 }
             });
-
-            // Width and height of palette canvas
-            const displayWidth = 16 * scaleX;
-            const displayHeight = 16 * scaleY;
-
             // Construct view
             // Palette number display
-            const palNumEl = util.createElement({
-                content: `${curPal+1}/${numPals}`,
-                appendTo: container
+            const paletteNumberElement = util.createElement({
+                content: `${currentPaletteIndex+1}/${numPalettes}`,
+                appendTo: innerContainer
             });
             // Palette canvas
-            const palCanvas = util.createElement({
+            const canvas = util.createElement({
                 tag: "canvas",
-                appendTo: container
+                appendTo: innerContainer,
+                width: 16 * scaleX, // 16 columns
+                height: 16 * scaleY, // 16 rows
             });
+            const context = canvas.getContext("2d") as CanvasRenderingContext2D;
             // Palette navigator
             util.createElement({
                 content: [prevArrow, nextArrow],
                 class: "lump-view-playpal-nav",
-                appendTo: container
+                appendTo: innerContainer
             });
-            palCanvas.width = displayWidth;
-            palCanvas.height = displayHeight;
-
-            // Draw palette to palCanvas
-            function drawPalette(palNum: number) {
+            // Draw palette to canvas
+            function drawPalette(paletteNum: number) {
                 if (!lump.data) {
                     return;
                 }
-
-                const context = palCanvas.getContext("2d") as CanvasRenderingContext2D;
-                for (let palIdx = 0; palIdx < 256; palIdx++) {
-
+                for (let colorIndex = 0; colorIndex < 256; colorIndex++) {
                     // Set RGB components
-                    const rgb = playpal.getColorBGRA(palNum, palIdx);
-
+                    const rgb = playpal.getColorBGRA(paletteNum, colorIndex);
                     // Calculate position to draw
-                    const column = palIdx % 16;
-                    const row = Math.floor(palIdx / 16);
+                    const column = (colorIndex % 16) * scaleX;
+                    const row = (Math.floor(colorIndex / 16)) * scaleY;
+                    // Set color and draw
+                    // Convert to hexadecimal string and trim alpha
+                    // HTML/CSS colors are in #RRGGBB format
+                    const rgbHex = rgb.toString(16).substring(2).padEnd(6, "0");
+                    context.fillStyle = `#${rgbHex}`;
+                    context.fillRect(column, row, scaleX, scaleY);
+                    // Update palette number display
+                    paletteNumberElement.innerText = (
+                        `${currentPaletteIndex+1}/${numPalettes}`);
```

Comment:
Style guide nitpick: Please put the closing parenthese on the following line, at the same indentation level as the line which has the corresponding open parenthese